src/main/java/org/opensearch/plugin/insights/core/service/categorizer/QueryShapeGenerator.java

```diff
@@ -257,85 +226,54 @@ static String buildFieldDataString(List<Function<Object, String>> methods, Named
-        }
+        private static final String INVALID_METHOD_EXCEPTION_STRING = "Invalid method referenced during field data map creation";
 
         /**
-         * Returns a map where the keys are query builders, and the values are lists of
-         * functions that extract field values from instances of these classes.
+         * Returns a map where the keys are query builder classes, and the values are lists of
+         * Methods that extract field data from instances of these classes.
          *
-         * @return a map with class types as keys and lists of field extraction functions as values.
+         * @return a map with class types as keys and lists of Methods as values.
          */
-        private static Map<Class<?>, List<Function<Object, String>>> getQueryFieldDataMap() {
-            return Map.ofEntries(
-                createEntry(AbstractGeometryQueryBuilder.class, AbstractGeometryQueryBuilder::fieldName),
-                createEntry(CommonTermsQueryBuilder.class, CommonTermsQueryBuilder::fieldName),
-                createEntry(ExistsQueryBuilder.class, ExistsQueryBuilder::fieldName),
-                createEntry(FieldMaskingSpanQueryBuilder.class, FieldMaskingSpanQueryBuilder::fieldName),
-                createEntry(FuzzyQueryBuilder.class, FuzzyQueryBuilder::fieldName),
-                createEntry(GeoBoundingBoxQueryBuilder.class, GeoBoundingBoxQueryBuilder::fieldName),
-                createEntry(GeoDistanceQueryBuilder.class, GeoDistanceQueryBuilder::fieldName),
-                createEntry(GeoPolygonQueryBuilder.class, GeoPolygonQueryBuilder::fieldName),
-                createEntry(MatchBoolPrefixQueryBuilder.class, MatchBoolPrefixQueryBuilder::fieldName),
-                createEntry(MatchQueryBuilder.class, MatchQueryBuilder::fieldName),
-                createEntry(MatchPhraseQueryBuilder.class, MatchPhraseQueryBuilder::fieldName),
-                createEntry(MatchPhrasePrefixQueryBuilder.class, MatchPhrasePrefixQueryBuilder::fieldName),
-                createEntry(MultiTermQueryBuilder.class, MultiTermQueryBuilder::fieldName),
-                createEntry(PrefixQueryBuilder.class, PrefixQueryBuilder::fieldName),
-                createEntry(RangeQueryBuilder.class, RangeQueryBuilder::fieldName),
-                createEntry(RegexpQueryBuilder.class, RegexpQueryBuilder::fieldName),
-                createEntry(SpanNearQueryBuilder.SpanGapQueryBuilder.class, SpanNearQueryBuilder.SpanGapQueryBuilder::fieldName),
-                createEntry(SpanTermQueryBuilder.class, SpanTermQueryBuilder::fieldName),
-                createEntry(TermQueryBuilder.class, TermQueryBuilder::fieldName),
-                createEntry(TermsQueryBuilder.class, TermsQueryBuilder::fieldName),
-                createEntry(WildcardQueryBuilder.class, WildcardQueryBuilder::fieldName)
-            );
+        private static ConcurrentHashMap<Class<?>, List<Method>> getQueryFieldDataMap() {
```

Comment:
getQueryFieldDataMap, getAggFieldDataMap, and getSortFieldDataMap all follow a similar pattern. Can refactor the logic for the following 3 methods. 

Something similar to:
```
private static <T> ConcurrentHashMap<Class<? extends T>, List<Method>> createFieldDataMap(Class<? extends T> clazz, String methodName) {
    ConcurrentHashMap<Class<? extends T>, List<Method>> map = new ConcurrentHashMap<>();
    try {
        map.put(clazz, List.of(clazz.getMethod(methodName)));
    } catch (NoSuchMethodException e) {
        throw new RuntimeException(INVALID_METHOD_EXCEPTION_STRING, e);
    }
    return map;
}

private static ConcurrentHashMap<Class<?>, List<Method>> getQueryFieldDataMap() {
    return createFieldDataMap(WithFieldName.class, "fieldName");
}

private static ConcurrentHashMap<Class<?>, List<Method>> getAggFieldDataMap() {
    return createFieldDataMap(ValuesSourceAggregationBuilder.class, "field");
}

private static ConcurrentHashMap<Class<?>, List<Method>> getSortFieldDataMap() {
    return createFieldDataMap(FieldSortBuilder.class, "getFieldName");
}
```